reactor-aeron-core/src/main/java/reactor/aeron/MessagePublication.java

```diff
@@ -266,82 +269,120 @@ public String toString() {
 
-    private Publication publication;
-    private B buffer;
-    private DirectBufferHandler<B> bufferHandler;
-    private MonoProcessor<Void> promise;
+    private final DirectBufferHandler bufferHandler;
+    private final MessagePublication parent;
 
     private long start;
+    private boolean requested;
 
-    private static <B> PublishTask<B> newInstance(
-        Publication publication,
-        B buffer,
-        DirectBufferHandler<B> bufferHandler,
-        MonoProcessor<Void> promise) {
+    private final MonoProcessor<Void> onDispose = MonoProcessor.create();
 
-      PublishTask<B> task = new PublishTask<>();
+    private volatile Object buffer;
+    private volatile Throwable error;
+
+    PublisherProcessor(DirectBufferHandler bufferHandler, MessagePublication messagePublication) {
+      this.bufferHandler = bufferHandler;
+      this.parent = messagePublication;
+      addSelf();
+    }
 
-      task.publication = publication;
-      task.buffer = buffer;
-      task.bufferHandler = bufferHandler;
-      task.start = 0;
-      task.promise = promise;
-      return task;
+    @Override
+    public Mono<Void> onDispose() {
+      return onDispose;
     }
 
-    private long publish() {
-      if (promise.isDisposed()) {
-        return 1;
+    void request() {
+      if (requested || isDisposed()) {
+        return;
       }
 
-      if (start == 0) {
-        start = System.currentTimeMillis();
+      Subscription upstream = upstream();
+      if (upstream != null) {
+        requested = true;
+        upstream.request(1);
       }
+    }
 
-      int length = bufferHandler.estimateLength(buffer);
+    void onParentError(Throwable throwable) {
+      resetBuffer();
+      onDispose.onError(throwable);
+    }
+
+    void reset() {
+      resetBuffer();
 
-      if (length < publication.maxPayloadLength()) {
-        BufferClaim bufferClaim = bufferClaims.get();
-        long result = publication.tryClaim(length, bufferClaim);
-        if (result > 0) {
-          try {
-            MutableDirectBuffer directBuffer = bufferClaim.buffer();
-            int offset = bufferClaim.offset();
-            bufferHandler.write(directBuffer, offset, buffer, length);
-            bufferClaim.commit();
-          } catch (Exception ex) {
-            bufferClaim.abort();
-            throw Exceptions.propagate(ex);
-          }
+      if (isDisposed()) {
+        removeSelf();
+        if (error != null) {
+          onDispose.onError(error);
+        } else {
+          onDispose.onComplete();
         }
-        return result;
-      } else {
-        return publication.offer(bufferHandler.map(buffer, length));
+        return;
       }
+
+      start = 0;
+      requested = false;
     }
 
-    private boolean isTimeoutElapsed(Duration timeout) {
+    @Override
+    protected void hookOnSubscribe(Subscription subscription) {
+      // no-op
+    }
+
+    @Override
+    protected void hookOnNext(Object value) {
+      buffer = value;
```

Comment:
Please add check that upstream supports backpressure (our requests)

Reply:
I propose to create and issue in reactor aeron and assign this issue on me. In a meantime let's move on with relaese of the new version.